app/src/main/java/org/oppia/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -443,40 +485,84 @@ class StatePlayerRecyclerViewAssembler private constructor(
   ) {
     val hasPreviousButton = playerFeatureSet.backwardNavigation && hasPreviousState
     when {
       hasGeneralContinueButton && playerFeatureSet.forwardNavigation -> {
-        pendingItemList += ContinueNavigationButtonViewModel(
-          hasPreviousButton,
-          previousNavigationButtonListener,
-          fragment as ContinueNavigationButtonListener
-        )
+        if (!shouldSplit) {
+          pendingItemList += ContinueNavigationButtonViewModel(
+            hasPreviousButton,
+            previousNavigationButtonListener,
+            fragment as ContinueNavigationButtonListener
+          )
+        } else {
+          rightPendingItemList += ContinueNavigationButtonViewModel(
+            false,
+            previousNavigationButtonListener,
+            fragment as ContinueNavigationButtonListener
+          )
+          if (hasPreviousButton) {
+            pendingItemList += PreviousButtonViewModel(
+              previousNavigationButtonListener
+            )
+          }
+        }
       }
       canContinueToNextState && playerFeatureSet.forwardNavigation -> {
-        pendingItemList += NextButtonViewModel(
-          hasPreviousButton,
-          previousNavigationButtonListener,
-          fragment as NextNavigationButtonListener
-        )
+        if (!shouldSplit) {
+          pendingItemList += NextButtonViewModel(
+            hasPreviousButton,
+            previousNavigationButtonListener,
+            fragment as NextNavigationButtonListener
+          )
+        } else {
+          rightPendingItemList += NextButtonViewModel(
+            false,
+            previousNavigationButtonListener,
+            fragment as NextNavigationButtonListener
+          )
+          if (hasPreviousButton) {
+            pendingItemList += PreviousButtonViewModel(
+              previousNavigationButtonListener
+            )
+          }
+        }
       }
       stateIsTerminal -> {
         if (playerFeatureSet.replaySupport) {
-          pendingItemList += ReplayButtonViewModel(fragment as ReplayButtonListener)
+          if (!shouldSplit) {
+            pendingItemList += ReplayButtonViewModel(fragment as ReplayButtonListener)
+          } else {
+            rightPendingItemList += ReplayButtonViewModel(fragment as ReplayButtonListener)
+          }
         }
         if (playerFeatureSet.returnToTopicNavigation) {
-          pendingItemList += ReturnToTopicButtonViewModel(
-            hasPreviousButton, previousNavigationButtonListener,
-            fragment as ReturnToTopicNavigationButtonListener
-          )
+          if (!shouldSplit) {
+            pendingItemList += ReturnToTopicButtonViewModel(
+              hasPreviousButton, previousNavigationButtonListener,
+              fragment as ReturnToTopicNavigationButtonListener
+            )
+          } else {
+            rightPendingItemList += ReturnToTopicButtonViewModel(
+              false, previousNavigationButtonListener,
```

Comment:
Reformat this to 2 lines and add inline comment for boolean

Reply:
Done